src/app/services/downloads.service.ts

```diff
@@ -42,19 +47,37 @@ export class DownloadsService {
         let tmpExportsCache: ExportCache[] = [];
         let changeCounter: number = 0;
         let index: number = 0;
+        let newCheck: boolean = true;
         // check number of changes between (new) exports and (old) exportsCache
         // exports and exportsCache are ordered desc by exportId
+        let tmpIndex: number = 0;
         exports.forEach(exp => {
             tmpExportsCache.push({ exportId: exp.exportId, status: exp.status });
+
             if (this.exportsCache != null) {
-                if (this.exportsCache.length > index
-                    && exp.exportId == this.exportsCache[index].exportId) {
-                    if (exp.status != this.exportsCache[index].status) {
+                if (index < this.exportsCache.length) {
+                    // check if incoming export is new
```

Comment:
The whole cache for exports seems unnecessary. This should just expose the list as it is returned by the backend (maybe sorted by status). Then you can let angular handle the change detection in the UI